src/shared-components/forms/style.js

```diff
@@ -0,0 +1,124 @@
+import styled from '@emotion/styled';
+import { css } from '@emotion/core';
+
+import { style as TYPOGRAPHY_STYLE } from '../typography';
+import { COLORS, BOX_SHADOWS, SPACING, ANIMATION } from '../../constants';
+
+export const HelperList = styled.ul`
+  list-style-type: none;
+  margin: 8px 0;
+  min-height: 23px;
+`;
+
+export const ErrorItem = styled.li`
+  ${TYPOGRAPHY_STYLE.error};
+`;
+
+export const HintItem = styled.li`
+  ${TYPOGRAPHY_STYLE.caption};
+`;
+
+export const FieldContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  margin-bottom: ${SPACING.small};
+`;
+
+export const Label = styled.label`
+  ${TYPOGRAPHY_STYLE.label};
+
+  ${({ disabled }) => disabled && `color:${COLORS.purple30};`};
+`;
+
+const inputStyles = css`
+  ${TYPOGRAPHY_STYLE.body};
+  appearance: none;
+  background: ${COLORS.white};
+  border: 1px solid ${COLORS.border};
+  border-radius: 0;
+  color: ${COLORS.primaryTint1};
+  padding: 13px ${SPACING.small};
+  transition: border-color ${ANIMATION.defaultTiming};
+  width: 100%;
+  outline: none;
+
+  ~ ul ${HintItem} {
+    transition: opacity ${ANIMATION.defaultTiming};
+    max-height: 0 !important;
+    opacity: 0 !important;
+  }
+
+  &:active,
+  &:focus {
+    outline: none;
+    border-color: ${COLORS.primary};
+    box-shadow: ${BOX_SHADOWS.focusSecondary};
+
+    ~ ul ${HintItem} {
+      max-height: 50px !important;
+      opacity: 1 !important;
+    }
+  }
+
+  &::placeholder {
+    color: ${COLORS.primaryTint3};
+  }
+
+  &[disabled] {
+    background-color: ${COLORS.disabled};
+    color: ${COLORS.textDisabled};
+    cursor: not-allowed;
+    border: 1px solid ${COLORS.border};
+    box-shadow: none;
+  }
+`;
+
+export const Input = styled.input`
+  ${inputStyles};
+`;
+
+export const Textarea = styled.textarea`
+  ${inputStyles};
+  color: ${COLORS.primary};
+  display: block;
+  height: 100%;
+  line-height: ${SPACING.base};
+  margin: 0 auto;
+  max-width: 35rem;
+  padding: ${SPACING.small};
+  resize: none;
+  transition: border-color ${ANIMATION.defaultTiming};
+  width: 100%;
+`;
+
+const errorStyles = css`
+  position: relative;
+
+  svg {
+    display: block;
+    width: 16px;
+    height: 16px;
+    position: absolute;
+    margin-top: 20px;
+    right: 16px;
```

Comment:
16px should be a spacing constant

Reply:
- I understand that SPACING constant are used for padding/margin/gaps. 
- Dimension values may either use other constant (ICON_SIZE maybe?) or left untouched?
- there are special cases where we need to pixel perfect accomodate the icon (like top:20px above)

Having said that I did these changes:
- changed margin-top to top:20px (this is special case pixel perfect)
- width and height left untouched with 16px
- changed right:${SPACING.small}

Reply:
yup that sounds like exactly what I had in mind. Thanks!